### DISCUSSION TITLE
Support using Tailwind in the browser

### DISCUSSION THREAD

**Discussion by remcohaszing** [tailwindlabs/tailwindcss#9284] [Ideas]
Currently when using Tailwind in the browser, this requires stubbing dependencies and using internal functions. It would be nice browser users could just use the official Tailwind API without support for file discovery.

**Comment by smithliam** [comment]
Are you looking for this?
https://tailwindcss.com/docs/installation/play-cdn

**Reply by remcohaszing** [reply]
Thanks for your suggestion, but I’m looking to actually consume the same API that Tailwindcss exposes in Node.js, but in the browser. Everything could work in the browser, except resolving files.

**Reply by remcohaszing** [reply]
For reference, [this works](https://github.com/remcohaszing/monaco-tailwindcss/blob/v0.4.2/src/tailwindcss.worker.ts#L169-L178), but it requires the use of internal functions, a [special build script](https://github.com/remcohaszing/monaco-tailwindcss/blob/main/build.js), and [various stubs](https://github.com/remcohaszing/monaco-tailwindcss/tree/main/src/stubs).

**Comment by jenstornell** [comment]
As a complement you can use Gimli Tailwind extension for Google Chrome:

https://chrome.google.com/webstore/detail/gimli-tailwind/fojckembkmaoehhmkiomebhkcengcljl

**Comment by mhsdesign** [comment]
yes id like that too ;)

for reference what currenly is needed to build something like an own tailwind jit cdn

https://github.com/mhsdesign/jit-browser-tailwindcss

exacly what @remcohaszing said in https://github.com/tailwindlabs/tailwindcss/discussions/9284#discussioncomment-3633689
